core/camel-core-processor/src/main/java/org/apache/camel/processor/resume/ResumableCompletion.java

```diff
@@ -70,6 +74,34 @@ public void onComplete(Exchange exchange) {
 
     @Override
     public void onFailure(Exchange exchange) {
-        LOG.warn("Skipping offset update for due to failure in processing");
+        Exception e = exchange.getException();
+        Object offset = exchange.getMessage().getHeader(Exchange.OFFSET);
+
+        if (offset instanceof Resumable) {
+            Resumable<?, ?> resumable = (Resumable<?, ?>) offset;
+
+            String logMessage = String.format(
+                    "Skipping offset update with address '%s' and offset value '%s' due to failure in processing: %s",
+                    resumable.getAddressable(), resumable.getLastOffset().offset(), e.getMessage());
+
+            if (LOG.isDebugEnabled()) {
+                CamelLogger.log(LOG, loggingLevel, logMessage, e);
```

Comment:
We would actually have 2 logging levels

- LOG instance that are controlled by log4j properties
- loggingLevel which is EIP option

So I think you need to log with stacktraces if one of them is DEBUG, and hence you need to test both.

And then the `loggingLevel` you pass in to CamelLogger should be hardcoded to DEBUG so its always logged at that level.

Reply:
Yes

Reply:
Great, thanks for confirming. I'll adjust.